dataprofiler/profilers/profiler_options.py

```diff
@@ -991,15 +1000,37 @@ def _validate_helper(self, variable_path='StructuredOptions'):
             ('data_labeler', DataLabelerOptions),
             ('correlation', CorrelationOptions)
         ])
-
-        for column in self.properties:
+        properties = self.properties
+        properties.pop('null_values')
+        for column in (properties):
             if not isinstance(self.properties[column], prop_check[column]):
                 errors.append("{}.{} must be a(n) {}.".format(
                     variable_path, column, prop_check[column].__name__))
             else:
                 errors += self.properties[column]._validate_helper(
                     variable_path=(variable_path + '.' + column
                                    if variable_path else column))
+        if self.null_values:
+            import re
+            reskey = True
+            resvalue = True
+            if isinstance(self.null_values, dict):
+                for key in self.null_values:
+                    if not isinstance(key, str):
+                        reskey = False
+                        break
+                for value in self.null_values.values():
+                    if not (value == 0 or isinstance(value, re.RegexFlag)):
+                        resvalue = False
+                        break
+            else:
+                reskey = False
+                resvalue = False
+            if reskey == False or resvalue == False:
+                errors.append("{}.null_values must be either None or "
+                              "a dictionary that contains keys of str type "
+                              "and values == 0 or are instances of "
+                              "a re.RegexFlag".format(variable_path))
```

Comment:
I think we can match the other formats in other validates:
```python
if self.null_values is not None and not (
        isinstance(self.null_values, dict) or 
        all(isinstance(key, str) and 
            (isinstance(value, re.RegexFlag) or value == 0) 
            for key, value in self.null_values.items())):
    errors.append("{}.null_values must be either None or "
                  "a dictionary that contains keys of str type "
                  "and values == 0 or are instances of "
                  "a re.RegexFlag".format(variable_path))
```
1. it checks if it is not `None` 
2. then checks if it isn't a dict, 
3.  checks to make sure:
   a.  all of the keys are strings 
   b. values are (re.RegexFags or 0)